src/Monolog/KinesisHandler.php

```diff
@@ -26,27 +25,18 @@ class KinesisHandler extends AbstractProcessingHandler
     /**
      * KinesisHandler constructor.
      *
-     * @param KinesisClient $client
      * @param string $streamName
-     * @param int $level
+     * @param string $level
      * @param bool $bubble
      */
-    public function __construct(KinesisClient $client, string $streamName, int $level = Logger::INFO, bool $bubble = true)
+    public function __construct(string $streamName, string $level, bool $bubble = true)
     {
         parent::__construct($level, $bubble);
 
-        $this->client = $client;
+        $this->client = app(KinesisClient::class);
         $this->streamName = $streamName;
-    }
 
-    /**
-     * Gets the default formatter.
-     *
-     * @return FormatterInterface
-     */
-    protected function getDefaultFormatter()
-    {
-        return new KinesisFormatter();
+        $this->formatter = app(KinesisFormatter::class);
```

Comment:
Can we move this from the constructor and back to the `getDefaultFormatter` method - the `AbstractProcessingHandler` -> `FormattableHandlerTrait` exposes this for the use case of setting the formatter for a handler